contracts/peripheral/PoolProxy.sol

```diff
@@ -44,18 +46,6 @@ contract PoolProxy is DecimalMath {
         require(fyDai.isMature() != true, "YieldProxy: Only before maturity");
         require(dai.transferFrom(msg.sender, address(this), daiUsed), "YieldProxy: Transfer Failed");
 
-        // Allow the Treasury to take chai when posting
-        if (chai.allowance(address(this), treasury) < type(uint256).max) chai.approve(treasury, type(uint256).max);
-
-        // Allow Chai to take dai for wrapping
-        if (dai.allowance(address(this), address(chai)) < type(uint256).max) dai.approve(address(chai), type(uint256).max);
-
-        // Allow pool to take dai for minting
-        if (dai.allowance(address(this), address(pool)) < type(uint256).max) dai.approve(address(pool), type(uint256).max);
-
-        // Allow pool to take fyDai for minting
-        if (fyDai.allowance(address(this), address(pool)) < type(uint112).max) fyDai.approve(address(pool), type(uint256).max);
```

Comment:
Why can we remove this? Is it redundant?

Reply:
All the proxy approvals have been moved to the `*WithSignature` functions.

It's a bit less user-friendly, because `addLiquidity` will fail if the approvals are not there. To get the approvals done you need to call `addLiquidityWithSignature`.

You should call `addLiquidity` only if `addLiquidityCheck` returns `(true, true, true)`.

I can move the approval functions to their own public function (`addLiquidityApprove`), to make things a bit more usable.

Reply:
Made more usable with `addLiquidityApprove` and friends.